src/bolt/bolt_api.c

```
@@ -281,20 +286,33 @@ RedisModuleString *get_query
+		uint32_t buf_size = query_len + 4096;
+		char *prametrize_query = rm_malloc(buf_size);
 		int n = sprintf(prametrize_query, "CYPHER ");
 		for (int i = 0; i < params_count; i++) {
 			uint32_t key_len;
 			bolt_read_string_size(&client->msg_buf.read, &key_len);
 			char key[key_len];
 			bolt_read_string(&client->msg_buf.read, key);
+			// grow buffer if needed
+			if(n + key_len + 4096 > buf_size) {
+				buf_size = n + key_len + 8192;
+				prametrize_query = rm_realloc(prametrize_query, buf_size);
```

Cppcheck (2.20.0)

[error] Common realloc mistake

(memleakOnRealloc)